### DISCUSSION TITLE
Thermal Hydraulics: FlowChannel1Phase Oreintated at an angle?

### DISCUSSION THREAD

**Discussion by TheBEllis** [idaholab/moose#21519] [Q&A Modules: General] [answered]
Hi MOOSE team, I am currently trying to use the thermal hydraulics module to simplify some fluids problems. I was wondering if there is any way to use 1D flow channels (specifically FlowChannel1Phase) that are at an angle or incline, as shown in the piece of art I have attached. If I try and change the orientation vector I get a reminder that these components should be aligned with one of the axes. It may be that there is no way round this currently. If so, is there any chance it will be added in future? Many thanks for any help.

![thExample](https://user-images.githubusercontent.com/54721825/177620264-2805f882-dd36-40bb-aa88-a59d501f9d8e.png)
)

**Comment by GiudGiud** [comment, marked as answer]
@licharlot

**Reply by licharlot** [reply]
There is no general restriction on the flow channels orientation. How are you connecting your pipes together?Are they coupled to a heat structure? Are you using any user objects like `LayeredAverage`?

**Reply by TheBEllis** [reply]
Ahhh thank you! They are coupled to a heat structure, I've been using HeatStructureFromFile3D and HeatTransferFromHeatStructure3D1Phase. I can see the restriction is on this side of things, as mentioned in the HeatTransferFromHeatStructure3D1Phase syntax documentation. I assume one way around this could be to manually set the gravity vector to change the frame of reference, is there a simpler way around this that I should be aware of?

**Reply by licharlot** [reply]
Yes there is that restriction for the 3D heat structure. I think adjusting the gravity vector for each flow channel is the best way at this point. You won't have that restriction if you use 2D heat structures (cylindrical or plate).
